archaius2-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java

```diff
@@ -15,299 +15,251 @@
-    }
-
-    @Provides
-    @Singleton
     @LibrariesLayer
     CompositeConfig getLibrariesLayer() {
-        return new DefaultCompositeConfig();
+        return librariesLayer;
     }
     
     @Provides
     @Singleton
-    @RemoteLayer
-    CompositeConfig getRemoteLayer() {
-        return new DefaultCompositeConfig();
+    @Raw
+    Config getRawConfig() {
+        return rawConfig;
     }
     
     @Provides
     @Singleton
-    @RemoteLayer
-    Config getRemoteLayer(@RemoteLayer CompositeConfig config) {
-        return config;
-    }
-    
-    /**
-     * This is the main config loader for the application.
-     * 
-     * @param config
-     * @param archaiusConfiguration
-     * @param readers
-     * @return
-     */
-    @Provides
-    @Singleton
-    ConfigLoader getLoader(
-            @RootLayer            Config config,
-            ArchaiusConfiguration archaiusConfiguration,
-            Set<ConfigReader>     readers
-            ) {
-        return DefaultConfigLoader.builder()
-                .withConfigReader(readers)
-                .withDefaultCascadingStrategy(archaiusConfiguration.getCascadeStrategy())
-                .withFailOnFirst(false)
-                .withStrLookup(ConfigStrLookup.from(config))
-                .build();
-    }
-
-    /**
-     * The internal RootLayer is meant only to construct the override hierarchy and 
-     * not to load or initialize any of the layers.  Initialization of layers is done
-     * is getConfig()
-     * 
-     * @param archaiusConfiguration
-     * @param settableLayer
-     * @param overrideLayer
-     * @param applicationLayer
-     * @param librariesLayer
-     * @param defaultsLayer
-     * @return
-     * @throws ConfigException
-     */
-    @Provides
-    @Singleton
-    @RootLayer
-    Config getInternalConfig(
-            ArchaiusConfiguration archaiusConfiguration,
-            @RuntimeLayer             SettableConfig    settableLayer,
-            @RemoteLayer              Config            overrideLayer,
-            @ApplicationLayer         CompositeConfig   applicationLayer, 
-            @LibrariesLayer           CompositeConfig   librariesLayer,
-            @DefaultsLayer            SettableConfig    defaultsLayer) throws ConfigException {
-        Builder builder = DefaultCompositeConfig.builder()
-                .withConfig(RUNTIME_LAYER_NAME,              settableLayer)
-                .withConfig(REMOTE_LAYER_NAME,               overrideLayer)
-                .withConfig(SYSTEM_LAYER_NAME,               SystemConfig.INSTANCE)
-                .withConfig(ENVIRONMENT_LAYER_NAME,          EnvironmentConfig.INSTANCE);
-        
-        if (archaiusConfiguration.getApplicationOverride() != null) {
-            builder.withConfig(APPLICATION_OVERRIDE_LAYER_NAME, archaiusConfiguration.getApplicationOverride());
+    Config getConfig(ConfigLoader loader, Injector injector) throws Exception {
+        if (applicationOverrides != null) {
```

Comment:
You can set overrides in code by injecting 

@RuntimeLayer
SettableConfig config;